### PR TITLE
DEVAI-33: Update Pipeline Definitions

### PR DESCRIPTION
**Pull request by Jdubrick**
This PR updates the pipeline definitions initially defined and used from: https://github.com/redhat-appstudio/tssc-sample-pipelines

In an effort to move away from RHTAP these updated definitions removed:
- TAS
- TPA
- ACS

As a result of this all of the tasks related to the gitops repo became redundant and as such were removed, leaving only the pipeline tasks for the source repo.

After this PR is merged then updates can be made to the software templates repo to reference these definitions instead of `tssc-sample-pipelines`. I have confirmed these definitions work with both `RHTAP` and our `AI-RHDH-Installer`.

To test you can reference my branch that has the changes pulled in here: https://github.com/Jdubrick/ai-lab-template/blob/DEVAI-33/all.yaml. Since changes need to be made to the namespaces for it to work completely with rhtap vs ai-rhdh-installer that branch is setup for rhtap. All you need to do is install rhdh via rhtap and import that `all.yaml` file.

Disclaimer: Commit history is pretty large since it included all the initial commits from `tssc-sample-pipelines` because this branch was empty so it wasn't forked from them and linked, if it is too messy I can try to fix that.

### CODE REVIEW

File: README.md
Comment (thepetk) [inline README.md]: Better to have a link as user's might not be aware of tekton.
```suggestion
These pipelines are in standard [tekton](https://tekton.dev/docs/) format.
```

File: pac/source-repo/README.md
Comment (thepetk) [inline pac/source-repo/README.md]: Not sure how much we can edit the content tho
```suggestion
This pipeline is used to create containerfile based SSCS (Software Supply Chain Security) builds. The pipeline run by this runner will clone the source, build an image with SBOM (Software Bill of Materials), and attestations and push these to the users image registry.  
```

**Review by thepetk — COMMENTED** [review]
Left some very minor comments I'll try to test everything later.

File: pac/source-repo/README.md
Comment (Jdubrick) [reply pac/source-repo/README.md]: We can edit it as much as we want since as far as I understand we are diverging fully from rhtap so we own these definitions! I added your suggestions as part of https://github.com/redhat-ai-dev/rhdh-pipelines/pull/1/commits/7849f2ed4444ddc127f911786e37fd2dd3111e60

File: pac/tasks/git-clone.yaml
Comment (michael-valdron) [inline pac/tasks/git-clone.yaml:148]: I wonder if we should use alternative images instead of ones from `konflux-ci` to remove any dependency on the Konflux/RHTAP setup. I'm good with this being the image with these changes but just thought I'd share for perhaps a future task.

File: pac/tasks/init.yaml
Comment (michael-valdron) [inline pac/tasks/init.yaml]: Should this be our label?

File: pac/tasks/summary.yaml
Comment (michael-valdron) [inline pac/tasks/summary.yaml]: > Should this be our label?

Same here.

File: pac/tasks/update-deployment.yaml
Comment (michael-valdron) [inline pac/tasks/update-deployment.yaml]: I would revise the git email for the tekton task to use our own identity:
```suggestion
        git config --global user.email "redhat-ai-dev@noreply.com"
```

File: pac/tasks/git-clone.yaml
Comment (Jdubrick) [reply pac/tasks/git-clone.yaml:148]: Created https://issues.redhat.com/browse/DEVAI-173 to address the copying of these images to ur own supported repo and quay. Also updated the labels and git config email here: https://github.com/redhat-ai-dev/rhdh-pipelines/pull/1/commits/0077e274cf9be6c5f32c6e14f216537c43653944

**Review by thepetk — APPROVED** [review]
In general we did a thorough test with @Jdubrick for the PR.

LGTM as long as @michael-valdron comments are addressed :)

File: pac/tasks/update-deployment.yaml
Comment (maysunfaisal) [inline pac/tasks/update-deployment.yaml:36]: so this is the old update deployment?

Comment (maysunfaisal) [reply pac/tasks/update-deployment.yaml:36]: we should probably switch like the RHTAP task but we can do that after merging and installer has been delivered

**Review by maysunfaisal — APPROVED** [review]
one minor comment but lgtm

File: pac/tasks/update-deployment.yaml
Comment (Jdubrick) [reply pac/tasks/update-deployment.yaml:36]: Yeah I believe this way matches the current setup of the installer

**Comment by maysunfaisal** [comment]
@Jdubrick I would squash merge this

**Comment by Jdubrick** [comment]
> @Jdubrick I would squash merge this

@maysunfaisal yes I will be, should merge today just waiting on re-review from @michael-valdron but I already spoke with him and he is going to do the re-review alongside his namespace config change

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Pull request merged**
